lib/flapjack/data/entity_check.rb

```diff
@@ -649,6 +651,8 @@ def update_state(new_state, options = {})
 
           # Even if this isn't a state change, we need to update the current state
           # hash summary and details (as they may have changed)
+          multi.hset("check:#{@key}", 'initial_failure_delay', (initial_failure_delay || Flapjack::DEFAULT_INITIAL_FAILURE_DELAY))
+          multi.hset("check:#{@key}", 'repeat_failure_delay', (repeat_failure_delay || Flapjack::DEFAULT_REPEAT_FAILURE_DELAY))
           multi.hset("check:#{@key}", 'summary', (summary || ''))
```

Comment:
Event senders are likely to retain that configuration across events and not simply send them on the first event submission. Because of this, I think a more intuitive behavior would be to say that removal of this property in Nagios/Sensu/etc. corresponds to the event using the system default, rather than some arbitrary stored value. (unset == default) makes more sense to me than (unset == undefined).